### PR TITLE
[WIP] Add zookeeper state store

### PR DESCRIPTION
**Pull request by flier**
# Description

Adds zookeeper as state store

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by yaron2** [comment]
Hey @flier thanks for this!

Can you run `make lint` and fix the linter comments?

File: state/zookeeper/zk.go [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+	return &config{
+		servers:           strings.Split(props.Servers, ","),
+		sessionTimeout:    sessionTimeout,
+		maxBufferSize:     maxBufferSize,
+		maxConnBufferSize: maxConnBufferSize,
+		keyPrefixPath:     props.KeyPrefixPath,
+	}, nil
+}
+
+type Conn interface {
```
Comment (yaron2) [inline state/zookeeper/zk.go]: What's the purpose of this interface? from what I see Conn is used from the `zk` package.

Comment (flier) [reply state/zookeeper/zk.go]: The `Conn` interface was used to generate `MockConn` for testing

**Comment by flier** [comment]
> Hey @flier thanks for this!
> 
> Can you run `make lint` and fix the linter comments?

File: state/zookeeper/zk.go (hunk 1 above)
Comment (yaron2) [reply state/zookeeper/zk.go]: So can this interface move to `zk_test.go`?

**Comment by yaron2** [comment]
@flier please resolve the conflicts and address the `Conn` interface.. once that's done we can merge. I'm excited to see this because we can maybe build on Zookeeper for leader election capabilities.

File: state/zookeeper/zk.go (hunk 1 above)
Comment (flier) [reply state/zookeeper/zk.go]: The problem is [zk.Conn](https://godoc.org/github.com/samuel/go-zookeeper/zk#Conn) is a structure which hard to be test without a real zookeeper instance. That's why I extract a `Conn` interface for testing.

**Comment by yaron2** [comment]
Thanks @flier !

**Pull request merged**